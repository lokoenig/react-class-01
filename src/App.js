import React, { useContext } from 'react';
import './App.css';
import { Outlet, Navigate } from "react-router-dom";
import NaviLinks  from "./components/navLinks";
import PageHeader from "./components/Header";
import { UserStatus, LoggedInContext } from './actions/firebase-auth';


function App() {
  const loggedIn = useContext(LoggedInContext);
  if (!loggedIn){
    return <Navigate replace to="/login" />
  } else {
  return (
    <div className="App">
      <PageHeader />
      <div className='sidebar' >
        <NaviLinks />
      </div>
      <div id="page-main">
        <Outlet />
        <UserStatus />
      </div>
     
    </div>
  );
  }
}

export default App;
