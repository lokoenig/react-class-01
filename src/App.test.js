import React from "react";
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from "react-router-dom";
import configureStore from 'redux-mock-store' //ES6 modules
import thunk from 'redux-thunk';

import App from './App';
import { Provider } from "react-redux";


const mockStore = configureStore([thunk]);
const store = mockStore({});


test('App framework changes', () => {
    const { getByText, asFragment } = render(
        <Provider store={store}>
        <BrowserRouter>
            <App />
        </BrowserRouter>
        </Provider>
        );
    //    screen.debug();
    expect(asFragment()).toMatchSnapshot();
});
