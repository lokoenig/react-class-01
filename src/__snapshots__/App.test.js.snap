// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App framework changes 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="sidebar"
    >
      <div
        class="navlink"
      >
        <a
          class=""
          href="/home"
        >
          Home
        </a>
        <a
          class=""
          href="/create"
        >
          Create
        </a>
        <a
          class=""
          href="/edit"
        >
          Edit
        </a>
        <a
          class=""
          href="/help"
        >
          Help
        </a>
        <a>
          Log Out
        </a>
      </div>
    </div>
    <div
      id="page-main"
    />
  </div>
</DocumentFragment>
`;
